src/index.js

```diff
@@ -26,6 +26,18 @@ const sleep = (ms) => new Promise(resolve => setTimeout(resolve, ms));
 
 export async function run() {
   try {
+    // custom working directory
+    const workingDirectoryInput = core.getInput('working-directory');
+    if (workingDirectoryInput) {
+      core.info(`Starting with custom working directory: ${workingDirectoryInput}`);
+    }
+    const workingDirectory = !workingDirectoryInput ? undefined : workingDirectoryInput;
```

Comment:
How about naming the variable `workingDirectory` instead of `workingDirectoryInput` in the first place? There is no need for two variables when a single one can do the job just as well, or even better.